### PR TITLE
:tada: Source Amazon Ads -  add sponsored_display_budget_rules stream

### LINKED ISSUE #26337: Source Amazon Ads: new stream BudgetRules
### What area the feature impact?

Connectors

### Revelant Information

https://advertising.amazon.com/API/docs/en-us/sponsored-products/budget-rules/overview
in version 3, if a campaign's budget has been impacted by a budget rule, you will see an effectiveBudget field included in the campaign budget object returned by [POST /sp/campaigns/list](https://advertising.amazon.com/API/docs/en-us/sponsored-products/3-0/openapi/prod#/Campaigns/ListSponsoredProductsCampaigns).

### PR DESCRIPTION
**Pull request by midavadim**
## What
Add sponsored_display_budget_rules stream

Issue: https://github.com/airbytehq/airbyte/issues/26337
Doc: https://advertising.amazon.com/API/docs/en-us/sponsored-display/3-0/openapi/prod#/BudgetRules/GetSDBudgetRulesForAdvertiser

## Recommended reading order
1. `x.java`
2. `y.python`

## 🚨 User Impact 🚨
No

### CODE REVIEW

**Comment by octavia-squidington-iii** [comment]
##  source-amazon-ads test report (commit [`0827b81682`](https://github.com/airbytehq/airbyte/pull/27512/commits/0827b81682677674f55bfdc385c6bca26b853117)) - ✅

⏲️  Total pipeline duration: 201 seconds

| Step                                                                     | Result   |
|:-------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-amazon-ads/metadata.yaml | ✅        |
| Connector version semver check.                                          | ✅        |
| Connector version increment check.                                       | ✅        |
| QA checks                                                                | ✅        |
| Code format checks                                                       | ✅        |
| Connector package install                                                | ✅        |
| Build source-amazon-ads docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                               | ✅        |
| Acceptance tests                                                         | ✅        |

🔗 [View the logs here](https://github.com/airbytehq/airbyte/actions/runs/5324953003)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-amazon-ads test
```

**Comment by octavia-squidington-iii** [comment]
##  source-amazon-ads test report (commit [`f6372510f3`](https://github.com/airbytehq/airbyte/pull/27512/commits/f6372510f380bcce74a03b75b8f8982d87a8f8f7)) - ✅

⏲️  Total pipeline duration: 192 seconds

| Step                                                                     | Result   |
|:-------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-amazon-ads/metadata.yaml | ✅        |
| Connector version semver check.                                          | ✅        |
| Connector version increment check.                                       | ✅        |
| QA checks                                                                | ✅        |
| Code format checks                                                       | ✅        |
| Connector package install                                                | ✅        |
| Build source-amazon-ads docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                               | ✅        |
| Acceptance tests                                                         | ✅        |

🔗 [View the logs here](https://github.com/airbytehq/airbyte/actions/runs/5324974379)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-amazon-ads test
```

**Review by lazebnyi — CHANGES_REQUESTED** [review]
Please update PR description and fix conflicts.

**Comment by midavadim** [comment]
> Please update PR description and fix conflicts

Let's process with review approval, there are a lot of continuous changes to Amazon Ads connector.

**Review by lazebnyi — CHANGES_REQUESTED** [review]
Please add link to issue and to provider doc with endpoint description and still need fix conflicts.

**Comment by octavia-squidington-iii** [comment]
##  source-amazon-ads test report (commit [`4aad15af82`](https://github.com/airbytehq/airbyte/pull/27512/commits/4aad15af82ed6dc565807b27a24b261708acbdcc)) - ❌

⏲️  Total pipeline duration: 139 seconds

| Step                                                                     | Result   |
|:-------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-amazon-ads/metadata.yaml | ✅        |
| Connector version semver check.                                          | ✅        |
| Connector version increment check.                                       | ✅        |
| QA checks                                                                | ✅        |
| Code format checks                                                       | ✅        |
| Connector package install                                                | ✅        |
| Build source-amazon-ads docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                               | ✅        |
| Acceptance tests                                                         | ❌        |

🔗 [View the logs here](https://storage.googleapis.com/airbyte-ci-reports/airbyte-ci/connectors/test/pull_request/midavadim_26401-source-amazon-sponsored-display-budget-rules/1687522384/4aad15af82ed6dc565807b27a24b261708acbdcc/source-amazon-ads/2.0.1/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-amazon-ads test
```

**Comment by octavia-squidington-iii** [comment]
##  source-amazon-ads test report (commit [`4658198dcb`](https://github.com/airbytehq/airbyte/pull/27512/commits/4658198dcb5a22a7989d065d9d77e91182d5a6c3)) - ✅

⏲️  Total pipeline duration: 194 seconds

| Step                                                                     | Result   |
|:-------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-amazon-ads/metadata.yaml | ✅        |
| Connector version semver check.                                          | ✅        |
| Connector version increment check.                                       | ✅        |
| QA checks                                                                | ✅        |
| Code format checks                                                       | ✅        |
| Connector package install                                                | ✅        |
| Build source-amazon-ads docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                               | ✅        |
| Acceptance tests                                                         | ✅        |

🔗 [View the logs here](https://storage.googleapis.com/airbyte-ci-reports/airbyte-ci/connectors/test/pull_request/midavadim_26401-source-amazon-sponsored-display-budget-rules/1687524924/4658198dcb5a22a7989d065d9d77e91182d5a6c3/source-amazon-ads/2.0.1/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-amazon-ads test
```

**Review by lazebnyi — APPROVED** [review]

File: airbyte-integrations/connectors/source-amazon-ads/Dockerfile [hunk 1]
```diff
@@ -14,5 +14,5 @@ ENTRYPOINT ["python", "/airbyte/integration_code/main.py"]
 
 
 
-LABEL io.airbyte.version=2.0.0
+LABEL io.airbyte.version=2.0.1
```
Comment (lazebnyi) [inline airbyte-integrations/connectors/source-amazon-ads/Dockerfile:17]:
```suggestion
LABEL io.airbyte.version=2.2.0
```

File: docs/integrations/sources/amazon-ads.md [hunk 2]
```diff
@@ -94,6 +94,7 @@ Information about expected report generation waiting time you may find [here](ht
 
 | Version | Date       | Pull Request                                             | Subject                                                                                                         |
 |:--------|:-----------|:---------------------------------------------------------|:----------------------------------------------------------------------------------------------------------------|
+| 2.0.1   | 2023-06-20 | [25412](https://github.com/airbytehq/airbyte/pull/25412) | Add sponsored_display_budget_rules stream                                                                       |
```
Comment (lazebnyi) [inline docs/integrations/sources/amazon-ads.md:97]:
```suggestion
| 2.2.0   | 2023-06-20 | [25412](https://github.com/airbytehq/airbyte/pull/25412) | Add sponsored_display_budget_rules stream                                                                       |
```

**Review by lazebnyi — CHANGES_REQUESTED** [review]

**Review by lazebnyi — APPROVED** [review]
Approved, but need version change.

File: docs/integrations/sources/amazon-ads.md (hunk 2 above)
Comment (lazebnyi) [reply docs/integrations/sources/amazon-ads.md:97]: Need add stream to list in the doc.

**Pull request closed without merging**